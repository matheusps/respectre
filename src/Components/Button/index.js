import React from 'react'
import { string, oneOf, bool } from 'prop-types'

const getClass = param => !!param && `btn-${param}`

const Button = ({
  children,
  color,
  size,
  action,
  circle,
  active,
  loading,
  disabled,
  ...props
}) => (
  <button
    className={`
        btn
        ${getClass(color)}
        ${getClass(size)}
        ${action && 'btn-action'}
        ${circle && 's-circle'}
        ${loading && 'loading'}
        ${active && 'active'}
    `}
    disabled={disabled}
    {...props}
  >
    {children}
  </button>
)

Button.propTypes = {
  color: string,
  size: oneOf(['sm', 'lg']),
  action: bool,
  circle: bool,
  active: bool,
  disabled: bool,
  loading: bool,
}

export default Button
