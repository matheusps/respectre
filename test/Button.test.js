import React from 'react'
import Button from '../src/Components/Button'
import renderer from 'react-test-renderer'

test('Button Matches Snapshot', () => {
  const component = renderer.create(<Button size="lg">Button Test</Button>)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
