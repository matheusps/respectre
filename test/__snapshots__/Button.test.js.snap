// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matches Snapshot 1`] = `
<button
  className="
        btn
        false
        btn-lg
        undefined
        undefined
        undefined
        undefined
    "
>
  Button Test
</button>
`;
